### PR TITLE
jd-cli: init at 1.2.1

### PR DESCRIPTION
**Pull request by Majiir**
###### Description of changes

Adds [**jd-cli**](https://github.com/intoolswetrust/jd-cli), a CLI around the `jd-core` Java decompiler which also powers the [`jd-gui`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/jd-gui/default.nix) package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Majiir** [comment]
I put this in `pkgs/tools/security` because that's where `jd-gui` lives. Other decompilers like `jadx` live there, but others like `cfr` are in `pkgs/development/tools`. The latter makes more sense to me, but I'd appreciate input from a more experienced maintainer.

**Comment by asbachb** [comment]
Validated build an binary working on amd64.
```console
/jd-cli --version
jd-cli version 1.2.1 - Copyright (C) 2015-2021 Josef Cacek
```

**Comment by onny** [comment]
Looks quite good, thank you! Why do you have most of the definitions outside of `stdenv.mkDerivation`? Looks a bit wired

**Comment by Majiir** [comment]
> Why do you have most of the definitions outside of `stdenv.mkDerivation`?

It's a pattern I've seen elsewhere in `nixpkgs` for using a fixed-output derivation for the dependencies from Maven. The `src`, `pname` and `version` are required for both `deps` and the main derivation.

Would it be better to move `deps` into the main derivation? I suppose that would make overlays easier to write.

**Comment by onny** [comment]
> > Why do you have most of the definitions outside of `stdenv.mkDerivation`?
> 
> It's a pattern I've seen elsewhere in `nixpkgs` for using a fixed-output derivation for the dependencies from Maven. The `src`, `pname` and `version` are required for both `deps` and the main derivation.
> 
> Would it be better to move `deps` into the main derivation? I suppose that would make overlays easier to write.

If this is a Meven specific pattern then I guess it's fine :)

File: pkgs/tools/security/jd-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    repo = pname;
+    rev = "${pname}-${version}";
+    hash = "sha256-rRttA5H0A0c44loBzbKH7Waoted3IsOgxGCD2VM0U/Q=";
+  };
+
+  deps = stdenv.mkDerivation {
+    name = "${pname}-deps";
+    inherit src;
+
+    buildInputs = [ maven ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/jd-cli/default.nix:18]: This should be nativeBuildInputs.

File: pkgs/tools/security/jd-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  src = fetchFromGitHub {
+    owner = "intoolswetrust";
+    repo = pname;
+    rev = "${pname}-${version}";
+    hash = "sha256-rRttA5H0A0c44loBzbKH7Waoted3IsOgxGCD2VM0U/Q=";
+  };
+
+  deps = stdenv.mkDerivation {
+    name = "${pname}-deps";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/jd-cli/default.nix:15]: This should be pname+version.

File: pkgs/tools/security/jd-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    dontFixup = true;
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-5d3ZLuzoEkPjh01uL/BuhJ6kevLdsm1P4PMLkEWaVUM=";
+  };
+
+in stdenv.mkDerivation rec {
+  inherit pname version src;
+
+  buildInputs = [ maven makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/jd-cli/default.nix:41]:
```suggestion
  nativeBuildInputs = [ maven makeWrapper ];
```

**Pull request merged**